net/retriever/retriever.go

```diff
@@ -67,9 +68,11 @@ type offlineMessage struct {
 }
 
 func NewMessageRetriever(cfg MRConfig) *MessageRetriever {
+	dial := gonet.Dial
 	var client *http.Client
 	if cfg.Dialer != nil {
-		tbTransport := &http.Transport{Dial: cfg.Dialer.Dial}
+		dial = cfg.Dialer.Dial
+		tbTransport := &http.Transport{Dial: dial}
```

Comment:
These transport changes shouldn't be repeated. Maybe rebase just the converter changes against lastest ethereum-master to ensure changes aren't leaking in.

Reply:
✅